mlrun/api/api/endpoints/datastore_profile.py

```diff
@@ -96,11 +96,27 @@ async def list_datastore_profiles(
         mlrun.common.schemas.AuthorizationAction.read,
         auth_info,
     )
-    return await run_in_threadpool(
+    profiles = await run_in_threadpool(
         mlrun.api.utils.singletons.db.get_db().list_datastore_profiles,
         db_session,
         project_name,
     )
+    if len(profiles) == 0:
+        return profiles
+    profile_names = [profile.name for profile in profiles]
+    allowed_profile_names = await mlrun.api.utils.auth.verifier.AuthVerifier().filter_project_resources_by_permissions(
+        mlrun.common.schemas.AuthorizationResourceTypes.datastore_profile,
+        profile_names,
+        lambda profile_name: (
+            project_name,
+            profile_name,
+        ),
+        auth_info,
+    )
+    filtered_data = [
+        profile for profile in profiles if profile.name in allowed_profile_names
+    ]
```

Comment:
You should directly filter the resources, no need to extract the names first, then filter them and then filter the objects based on filtered names. Simply do something like:
```
filtered_data = await mlrun.api.utils.auth.verifier.AuthVerifier().filter_project_resources_by_permissions(
    mlrun.common.schemas.AuthorizationResourceTypes.datastore_profile,
    profiles,
    lambda profile: ( project_name, profile.name ),
    auth_info,
)
```